src/vivarium_census_prl_synth_pop/data/loader.py

```diff
@@ -305,6 +308,58 @@ def load_address_data(key: str, location: str) -> pd.DataFrame:
     return df_deepparse_address_data
 
 
+def load_ssn_data(key: str, location: str) -> pd.DataFrame:
+    """Generate a randomly shuffled dataframe of all possible SSNs where each of
+    the three columns holds the allowable parts of a valid id. SSNs take the
+    form of <area>-<group>-<serial> with the following acceptable ranges:
+        - area: 001-899, excluding 666
+        - group: 01-99
+        - serial: 0001-9999
+    """
+    # Generate allowable integers
+    area = list(range(1, 666)) + list(range(667, 900))
+    group = list(range(1, 100))
+    serial = list(range(1, 10000))
+
+    return _generate_randomized_id_df(area, group, serial)
+
+
+def load_itin_data(key: str, location: str) -> pd.DataFrame:
+    """Generate a randomly shuffled dataframe of all possible ITINs where each of
+    the three columns holds the allowable parts of a valid id. ITINs take the
+    form of <area>-<group>-<serial> with the following acceptable ranges:
+        - area: 900-999
+        - group: 50-65, 70-88, 90-92, 94-99
+        - serial: 0001-9999
+    """
+    # Generate allowable integers
+    area = list(range(900, 1000))
+    group = (
+        list(range(50, 66)) + list(range(70, 89)) + list(range(90, 93)) + list(range(94, 100))
+    )
+    serial = list(range(1, 10000))
+
+    return _generate_randomized_id_df(area, group, serial)
+
+
+#####################
+#  Helper functions #
+#####################
+
+
+def _generate_randomized_id_df(
+    area: List[int], group: List[int], serial: List[int]
+) -> pd.DataFrame:
+    """Given lists of allowable area, group, and serial integers, generate a
+    randomly sorted dataframe of the product of the three groups
+    """
+    parts = [(area, group, serial) for area, group, serial in product(area, group, serial)]
+    ids = pd.DataFrame(data=parts, columns=["area", "group", "serial"])
+    ids = ids.sample(frac=1).reset_index(drop=True)
```

Comment:
You should provide a seed here so we are guaranteed to get the same artifact every time we rebuild it.

Reply:
Whoops, good catch